src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -262,63 +344,97 @@ class npc_mograine : public CreatureScript
-                case 2:
-                    me->SetSheath(SHEATH_STATE_UNARMED);
-                    me->SetStandState(UNIT_STAND_STATE_KNEEL);
-                    return 2 * IN_MILLISECONDS;
-                case 3:
-                    Talk(3);
-                    return 10 * IN_MILLISECONDS;
-                case 4:
-                    me->SummonCreature(NPC_HIGHLORD_MOGRAINE, 1065.130737f, 1399.350586f, 30.763723f, 6.282961f, TEMPSUMMON_TIMED_DESPAWN, 400000)->SetName("Highlord Mograine");
-                    me->FindNearestCreature(NPC_HIGHLORD_MOGRAINE, 200.0f)->SetFaction(FACTION_FRIENDLY);
-                    return 30 * IN_MILLISECONDS;
-                case 5:
-                    mograine->StopMovingOnCurrentPos();
-                    mograine->AI()->Talk(0);
-                    mograine->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
-                    return 4 * IN_MILLISECONDS;
-                case 6:
-                    me->SetStandState(UNIT_STAND_STATE_STAND);
-                    return 2 * IN_MILLISECONDS;
-                case 7:
-                    Talk(4);
-                    return 4 * IN_MILLISECONDS;
-                case 8:
-                    mograine->AI()->Talk(1);
-                    return 11 * IN_MILLISECONDS;
-                case 9:
-                    mograine->HandleEmoteCommand(EMOTE_ONESHOT_BATTLE_ROAR);
-                    return 4 * IN_MILLISECONDS;
-                case 10:
-                    me->SetSheath(SHEATH_STATE_UNARMED);
-                    me->SetStandState(UNIT_STAND_STATE_KNEEL);
-                    Talk(5);
-                    return 2 * IN_MILLISECONDS;
-                case 11:
-                    mograine->CastSpell(me, SPELL_FORGIVENESS, false);
-                    return 1 * IN_MILLISECONDS;
-                case 12:
-                    mograine->CastSpell(me, SPELL_COSMETIC_CHAIN, true);
-                    return 0.5 * IN_MILLISECONDS;
-                case 13:
-                    mograine->AI()->Talk(2);
-                    mograine->DespawnOrUnsummon(3 * IN_MILLISECONDS);
-                    mograine->Kill(me, me, true);
-                    return 0;
-                default:
-                    if(mograine)
-                        mograine->DespawnOrUnsummon(0);
-                    return 0;
+            case EVENT_MOGRAINE_FACING_PLAYER:
+                me->SetFacingToObject(playerWhoStartedAshbringer);
+                events.ScheduleEvent(EVENT_MOGRAINE_KNEEL, 1s, 3s);
+                break;
+            case EVENT_MOGRAINE_KNEEL:
+                me->SetSheath(SHEATH_STATE_UNARMED);
+                me->SetStandState(UNIT_STAND_STATE_KNEEL);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK3, 1s, 2s);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK3:
+                me->HandleEmoteCommand(EMOTE_ONESHOT_TALK);
+                me->AI()->Talk(SAY_MO_AB_TALK3, playerWhoStartedAshbringer);
```

Comment:
Talk is contained in Creature AI so you can just do `Talk(SAY_MO_AB_TALK3, playerWhoStartedAshbringer);` when it is me talking. Also you can add emotes to creature_text